app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,55 @@
+class TeachersController < ApplicationController
+
+  def index
+    @teachers = Teacher.all
+  end
+
+  def new
+    @teacher = Teacher.new
+  end
+
+  def create
+    @teacher = Teacher.new(teacher_params)
+
+    @teacher.save
+    redirect_to teachers_path
+  end
+
+  def teacher_params
+    params.require( :teacher).permit( :name)
+  end
+
+  def edit
+    @teacher = Teacher.find(teacher_params[:id])
```

Comment:
This would just be `params[:id]`. `teacher_params` filters out `params`. When you try to load this action, it will throw an error.